ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp

```diff
@@ -165,10 +165,10 @@ void AllGather::validate(const std::vector<Tensor> &input_tensors) const {
     }
 }
 
-std::vector<tt::tt_metal::LegacyShape> AllGather::compute_output_shapes(const std::vector<Tensor> &input_tensors) const {
-    auto shape = input_tensors[0].get_legacy_shape();
+std::vector<ttnn::SimpleShape> AllGather::compute_output_shapes(const std::vector<Tensor> &input_tensors) const {
+    auto shape = input_tensors[0].get_padded_shape(); // TODO: Replace with get_logical_shape()
```

Comment:
This might hide more bugs later. Why does switching to `logical_shape` not work here?

Reply:
using logical shape throws error in layernorm op due to padding

Reply:
What's the padding? I think it would be good to fix both in this PR. The workaround here only works because allgather itself doesn't have padded tests (i am assuming)

Reply:
What's the test to repro? I can take a look at this as well

Reply:
`tests/ttnn/unit_tests/operations/test_distributed_layernorm.py`
This is the test file that is failing if we use logical shape

Reply:
The problem is:
1) All-gather doesn't support padding properly yet and when distributed layernorm was implemented it knew this and as a sort of hack - it accounted for this behaviour.
2) Fixing this properly in all-gather (with acceptable perf) is a non-trivial amount of work. Until then, this PR will migrate us to new tensor without regressing on old functionality.

Reply:
Ok merge it for now. Would distributed layernorm be safe to flip to `SimpleShape` as well? Or does it depend on some behaviour in all gather?

Reply:
it's the same problem - it was built knowing this limitation of all-gather there so when we fix it in all-gather we need to make sure to flip it there too.